test/over_react/component/shared_error_boundary_tests.dart

```diff
@@ -304,6 +336,72 @@ void sharedErrorBoundaryTests(BuilderOnlyUiFactory builder) {
+            setUp(() async {
+              sharedSetup(errorBoundaryChildren: FlawedOnMount()());
+            });
+
+            test('the components wrapped by the ErrorBoundary get unmounted', () {
+              expect(getByTestId(jacket.getInstance(), 'flawedComponent'), isNull,
+                  reason: 'The flawed component should have been unmounted');
+              expect(jacket.getDartInstance().state.showFallbackUIOnError, isTrue,
+                  reason: 'Fallback UI should be rendered instead of the flawed component tree');
+              expect(jacket.getNode(),
+                  hasAttr(defaultTestIdKey, 'ErrorBoundary.unrecoverableErrorInnerHtmlContainerNode'));
+              calls.clear();
+            });
+
+            test('the correct callbacks are called with the correct arguments', () {
+              expect(calls.any((call) => call.keys.single == 'onComponentIsUnrecoverable'), isTrue,
+                  reason: 'onComponentIsUnrecoverable should have been called');
+
+              _setCallbackVarValues();
+
+              expect(errorSentToComponentDidCatchCallback, isA<FlawedOnMountComponentException>());
+              expect(errorSentToComponentIsUnrecoverableCallback, isA<FlawedOnMountComponentException>());
+              expect(errorSentToComponentDidCatchCallback, errorSentToComponentIsUnrecoverableCallback);
+
+              expect(errorInfoSentToComponentDidCatchCallback, isA<ReactErrorInfo>());
+              expect(errorInfoSentToComponentIsUnrecoverableCallback, isA<ReactErrorInfo>());
+              expect(errorInfoSentToComponentDidCatchCallback, errorInfoSentToComponentIsUnrecoverableCallback);
+            });
+          });
+
+          group('and they occurred on every render and fallbackUIRenderer is set', () {
+            setUp(() async {
+              sharedSetup(
+                errorBoundaryChildren: FlawedOnMount()(),
+                errorBoundaryProps: (ErrorBoundaryPropsMapView({})
+                    ..fallbackUIRenderer = (_, __) => (Dom.div()..addTestId('fallbackUIRenderer'))()
+                  ),
+              );
+            });
+
+            test('the components wrapped by the ErrorBoundary get unmounted', () {
+              expect(getByTestId(jacket.getInstance(), 'flawedComponent'), isNull,
+                  reason: 'The flawed component should have been unmounted');
+              expect(jacket.getDartInstance().state.showFallbackUIOnError, isTrue,
+                  reason: 'Fallback UI should be rendered instead of the flawed component tree');
+              expect(jacket.getNode(),
+                  hasAttr(defaultTestIdKey, 'fallbackUIRenderer'));
+              calls.clear();
+            });
+
+            test('the correct callbacks are called with the correct arguments', () {
+              expect(calls.any((call) => call.keys.single == 'onComponentDidCatch'), isTrue,
+                  reason: 'onComponentDidCatch should have been called');
```

Comment:
It seems like the primary difference between this _"and they occurred on every render and fallbackUIRenderer is set"_ block - and the previous _"and they occurred on every render"_ block - is that the latter should not ever get to the "unrecoverable" state, correct?  If that is the case - I believe an assertion needs to be made that `props. onComponentIsUnrecoverable` is never called.

Reply:
The usage of `latter` is confusing because i am interpreting it as you are referring to the "latter" in the actually testing order not in the sentence you used. which would be the former? lol ugh, either way i think i understood and added an expect that its not there. you let me know if i misunderstood.

Reply:
mm yeah - i can see how that was confusing... i think you got it though.